lib/raven.interceptor.ts

```diff
@@ -1,63 +1,79 @@
+import { Handlers } from '@sentry/node';
+import { GqlArgumentsHost, GraphQLArgumentsHost } from '@nestjs/graphql';
 
 @Injectable()
 export class RavenInterceptor implements NestInterceptor {
-  
-  constructor(
-    private readonly options: IRavenInterceptorOptions = { context: 'Http' },
-  ) {
-  }
-  
-  intercept(
-    context: ExecutionContext,
-    next: CallHandler,
-  ): Observable<any> {
-    const http = context.switchToHttp();
-    const ws = context.switchToWs();
-    const rpc = context.switchToRpc();
-
-    // Default to Http as conteext
-    if (!this.options.context) this.options.context = 'Http';
+  constructor(private readonly options: IRavenInterceptorOptions = {}) {}
 
+  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
     // first param would be for events, second is for errors
-    return next.handle()
-    .pipe(
-      tap(null, (exception) => {
+    return next.handle().pipe(
+      tap(null, exception => {
         if (this.shouldReport(exception)) {
           Sentry.withScope(scope => {
-            // TODO: When https://github.com/nestjs/nest/issues/1581 gets implemented switch to that
-            switch(this.options.context){
-              case 'Http': return this.captureHttpException(scope as any, http, exception);
-              case 'Ws': return this.captureWsException(scope as any, ws, exception);
-              case 'Rpc': return this.captureRpcException(scope as any, rpc, exception);
-              case 'GraphQL': return this.captureGraphQLException(scope as any, GqlArgumentsHost.create(context), exception);
+            switch (context.getType()) {
+              case 'http':
+                return this.captureHttpException(
+                  scope as any,
+                  context.switchToHttp(),
+                  exception,
+                );
+              case 'ws':
+                return this.captureWsException(
+                  scope as any,
+                  context.switchToWs(),
+                  exception,
+                );
+              case 'rpc':
+                return this.captureRpcException(
+                  scope as any,
+                  context.switchToRpc(),
+                  exception,
+                );
+              default:
+                return this.captureGraphQLException(
+                  scope as any,
+                  GqlArgumentsHost.create(context),
+                  exception,
+                );
             }
           });
         }
-      })
+      }),
     );
   }
 
-  private captureGraphQLException(scope: Scope, gqlHost: GraphQLArgumentsHost, exception): void {
-
+  private captureGraphQLException(
+    scope: Scope,
+    gqlHost: GraphQLArgumentsHost,
+    exception,
+  ): void {
     // Same as HttpException
-    const data = Handlers.parseRequest(<any>{}, gqlHost.getContext(), this.options);
+    const data = Handlers.parseRequest(
+      <any>{},
+      gqlHost.getContext(),
```

Comment:
`gqlHost.getContext()` isn't the `req` object in nest by default, so it shouldn't be used as is here.
This caused missing request metadata in sentry until now when using `nest-raven` with graphql.
I suggest
```
const context = gqlHost.getContext();

const data = Handlers.parseRequest(
      <any>{},
      context.req || context,
      this.options,
    );
```
Since it is common to add a `req` field to the context while configuring nest-graphql, but it is by no means the default behaviour so the current code is provided as fallback.

Reply:
fixed :)